From 8217ae915378610aa033ff6fce1db9f575ca775d Mon Sep 17 00:00:00 2001
From: Jeremy Compostella <jeremy.compostella@intel.com>
Date: Fri, 13 May 2016 13:18:40 +0200
Subject: [PATCH 763/769] ANDROID: dm: Use generic bio chaining

dm-verity-target still uses bio_endio() for bio completions while it
does not artificially increase the bi_remaining counter.  It leads to
a systematic crash at boot time.

Use the generic mechanism for chaining bio completions introduced by
196d38bccfcfa32faed8c561868336fdfa0fe8e4.

Change-Id: I87e323eda57a1ee00416766d1fbb0770d15c0fda
Signed-off-by: Jeremy Compostella <jeremy.compostella@intel.com>
---
 drivers/md/dm-verity-target.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/md/dm-verity-target.c b/drivers/md/dm-verity-target.c
index 7fd6d20..c0331eb 100644
--- a/drivers/md/dm-verity-target.c
+++ b/drivers/md/dm-verity-target.c
@@ -467,7 +467,7 @@ static void verity_finish_io(struct dm_verity_io *io, int error)
 
 	verity_fec_finish_io(io, error);
 
-	bio_endio(bio, error);
+	bio_endio_nodec(bio, error);
 }
 
 static void verity_work(struct work_struct *w)
-- 
1.9.1

